Source/SessionManager.swift

```diff
@@ -261,15 +265,31 @@ open class SessionManager {
 
             return request
         } catch {
-            return request(failedWith: error)
+            return request(originalRequest, failedWith: error)
         }
     }
 
     // MARK: Private - Request Implementation
 
-    private func request(failedWith error: Error) -> DataRequest {
-        let request = DataRequest(session: session, requestTask: .data(nil, nil), error: error)
-        if startRequestsImmediately { request.resume() }
+    private func request(_ urlRequest: URLRequest?, failedWith error: Error) -> DataRequest {
+        var requestTask: Request.RequestTask = .data(nil, nil)
+
+        if let urlRequest = urlRequest {
+            let originalTask = DataRequest.Requestable(urlRequest: urlRequest)
+            requestTask = .data(originalTask, nil)
+        }
+
+        let isAdaptError = error is AdaptError
+        let error = error.extractedAdaptError
```

Comment:
It seems simpler to just treat `extractedAdaptError` as an optional, rather than having to check type and extract the value separately. Was there a reason for this pattern over just making the `extractedAdaptError` optional? Also, the `extractedAdaptError`, as it is now, seems misleading, because it's either the `extractedAdaptError` or the original error.